pkg/artifacts/artifacts_storage.go

```diff
@@ -68,11 +54,31 @@ func InitializeArtifactStorage(pr *v1alpha1.PipelineRun, c kubernetes.Interface)
 	return NewArtifactBucketConfigFromConfigMap(configMap)
 }
 
+func needsPVC(configMap *corev1.ConfigMap, err error) bool {
+	if err != nil {
+		return true
+	}
+	if configMap == nil {
+		return true
+	}
+	if configMap.Data == nil {
+		return true
+	}
+	if location, ok := configMap.Data[v1alpha1.BucketLocationKey]; !ok {
+		return true
+	} else {
+		if strings.TrimSpace(location) == "" {
+			return true
+		}
+	}
```

Comment:
i feel like in some of these cases we should be logging a warning or something, what do you think? if the user provided the configmap but didn't provide the data or url, that seems like they made a mistake? :thinking:

Reply:
@bobcatfish good point, I'll add some logging in there :wink: